### PR TITLE
Disable CA2022 error

### PR DESCRIPTION
**Pull request by rchauhan18**
Fix build failure: https://github.com/dotnet/wpf/pull/9163
Replacing System.IO.Stream.Read with System.IO.Stream.ReadExactly, to resolve the CA2022 error.

### CODE REVIEW

**Review by singhashish-wpf — DISMISSED** [review]
LGTM, newly added CA2022, disabling for now.

**Comment by rchauhan18** [comment]
@stephentoub 
I have added changes to fix CA2022 warning in WPF. Can you please review the changes?

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs [hunk 1]
```diff
@@ -189,7 +189,22 @@ internal static uint DecodeAsISF(Stream stream, GuidList guidList, uint maximumS
                     }
 
                     byte[] data = new byte[ropSize];
-                    stream.Read (data, 0, (int)ropSize);
+#if NET7_0_OR_GREATER
```
Comment (ThomasGoulet73) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs:192]: Why did you add `#if` ? This project should always build with NET7_0_OR_GREATER set to true so I don't understand the value of having the `#else`.

Comment (stephentoub) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs:192]: I'm curious as well... doesn't this only build for net9.0?
https://github.com/dotnet/wpf/blob/47efe4600a00ac059aeca109459e9a4e8132d907/Directory.Build.props#L5-L6

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs [hunk 2]
```diff
@@ -1586,7 +1586,22 @@ private int GetDataIntoOleStructsByTypeMedimIStream(string format, object data,
 
                         byte[] buffer = new byte[NativeMethods.IntPtrToInt32(size)];
                         inkStream.Position = 0;
-                        inkStream.Read(buffer, 0, NativeMethods.IntPtrToInt32(size));
+#if NET7_0_OR_GREATER
+                inkStream.ReadExactly(buffer, 0, NativeMethods.IntPtrToInt32(size));
```
Comment (stephentoub) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs:1590]: What kind of stream can inkStream be? If it can be an arbitrary stream, are there tests (or can we add tests) validating what happens when the stream provides back fewer bytes than requested?

Comment (stephentoub) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs:1590]:
```suggestion
                inkStream.ReadExactly(buffer, 0, buffer.Length);
```

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs [hunk 3]
```diff
@@ -189,7 +189,22 @@ internal static uint DecodeAsISF(Stream stream, GuidList guidList, uint maximumS
...
+                stream.ReadExactly(data, 0, (int)ropSize);
```
Comment (stephentoub) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs:193]:
```suggestion
                stream.ReadExactly(data, 0, data.Length);
```

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs [hunk 4]
```diff
@@ -1725,7 +1740,22 @@ private int SaveStreamToHandle(IntPtr handle, Stream stream, bool doNotReallocat
 
                 bytes = new byte[NativeMethods.IntPtrToInt32(size)];
                 stream.Position = 0;
-                stream.Read(bytes, 0, NativeMethods.IntPtrToInt32(size));
+#if NET7_0_OR_GREATER
+                stream.ReadExactly(bytes, 0, NativeMethods.IntPtrToInt32(size));
```
Comment (stephentoub) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs:1744]:
```suggestion
                stream.ReadExactly(bytes);
```

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs (hunk 1 above)
Comment (ashishkumarsinghh) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs:192]: +1, we will remove it. Our first try was to just disable it, then we checked the runtime PR and added same changes to call `ReadExactly`

Comment (singhashish-wpf) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/MS/internal/Ink/InkSerializedFormat/DrawingAttributeSerializer.cs:192]: +1, We will remove it. Our first try was to disable it, then we checked on the runtime changes and added the same changes.

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs (hunk 2 above)
Comment (stephentoub) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs:1590]: I typically do this by writing a type that derives from MemoryStream and overrides the Read methods to return back a maximum of one byte, e.g.
```C#
internal sealed class TrickleStream : MemoryStream
{
    public override int Read(byte[] buffer, int offset, int count) =>
        base.Read(buffer, offset, Math.Min(count, 1));
    ...
}
```

Comment (miloush) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs:1590]: Oh this particular one is actually from OLE, so in most cases it might be limited to IStream, although within application you can still store any Stream in the DataStore. This particular code doesn't have many checks around, unlike the deserialization. If Read returns 0, this just writes 0 to the medium and calls it a day. I can see how if we start throwing here, applications that did not use to crash can start crashing. Still probably a reasonably thing to do.

**Comment by rchauhan18** [comment]
Thanks @stephentoub for the suggestions.
I have updated the PR.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Review by singhashish-wpf — APPROVED** [review]
